crates/biome_js_analyze/src/semantic_analyzers/nursery/use_number_properties.rs

```diff
@@ -0,0 +1,192 @@
+    /// parseFloat("1.1"); // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// NaN; // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// Infinity; // true
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// -Infinity; // true
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// Number.parseInt("1"); // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.parseFloat("1.1"); // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.NaN; // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.POSITIVE_INFINITY; // false
+    /// ```
+    ///
+    /// ```js
+    /// Number.NEGATIVE_INFINITY; // false
+    /// ```
+    ///
+    pub(crate) UseNumberProperties {
+        version: "next",
+        name: "useNumberProperties",
+        recommended: true,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+impl Rule for UseNumberProperties {
+    type Query = Semantic<AnyJsExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let model = ctx.model();
+        let (reference, name) = global_identifier(node)?;
+        let names = ["parseInt", "parseFloat", "NaN", "Infinity"];
+        if !names.contains(&name.text()) {
+            return None;
+        }
+        model.binding(&reference).is_none().then_some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Use the "<Emphasis>"Number"</Emphasis>" properties instead of the global ones."
+                },
+            )
+            .note(markup! {
+                "ECMAScript 2015 moved globals onto the "<Emphasis>"Number"</Emphasis>" constructor for consistency and to slightly improve them."
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let mut mutation = ctx.root().begin();
+        let (old, new) = match node {
+            AnyJsExpression::JsIdentifierExpression(expression) => {
+                let name = expression.name().ok()?.text();
+                let replacement = match name.as_str() {
+                    "parseInt" => "parseInt",
+                    "parseFloat" => "parseFloat",
+                    "NaN" => "NaN",
+                    "Infinity" => {
+                        let is_negative = expression
+                            .syntax()
+                            .parent()
+                            .and_then(|parent| {
+                                parent.children_with_tokens().find_map(|child| match child {
+                                    biome_rowan::NodeOrToken::Token(token) => {
```

Comment:
We already import `biome_rowan`, so you don't need to use full path.

Reply:
Thanks for comment.
It seems that my being a beginner in Rust has been exposed, hasn't it? 😄

fix: 99e0270

Reply:
This makes me think that we could enable a rustc lint rule to check this.